apps/screenshot-diff/src/computeScreenshotDiff.ts

```diff
@@ -40,12 +41,16 @@ export const computeScreenshotDiff = async (
   const diffResultPath = join(tmpDir, "diff.png");
 
   await Promise.all([
-    download({
-      s3,
-      outputPath: baseScreenshotPath,
-      Bucket: bucket,
-      Key: screenshotDiff.baseScreenshot!.s3Id,
-    }),
+    ...(screenshotDiff.baseScreenshot
```

Comment:
This code is not well factored. It has to be rewritten. If we have several files that can be downloaded or not, then we have to manage them differently to be able to unlink them properly. We have to create an abstraction on the S3 file.

```js
const baseScreenshot = new S3Image({ bucket, key })

await baseScreenshot.download()
// Will failed if a `download` has not be done before
const baseScreenshotFilepath = baseScreenshot.getFilePath()
// Will unlink or do nothing depending if a dowload has been made or not
await baseScreenshot.unlink()
// Get image dimensions, cached
await baseScreenshot.getDimensions()
```

In the `imageDifference` we will use these file too, to benefit of cached of `getDimensions`.

Reply:
When you say that getFilePath should fail, do you mean throw an error or return `undefined` ?

Reply:
Throw because it should not be called if download is not called before, it would mean the code is incoherent.